src/averageAges.js

```diff
@@ -35,9 +39,17 @@ function calculateMenAverageAge(people, century) {
  * @param {boolean} withChildren - optional
  *
  * @return {number}
- */
+*/
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womans = people.filter(person =>
+    withChildren
+      ? people.find(child => child.mother === person.name)
+      : person.sex === 'f'
+  );
+
+  return womans.reduce((sumOfAge, { died, born }) => {
+    return sumOfAge + died - born;
+  }, 0) / womans.length;
```

Comment:
you are using almost an identical function for calculating the average age of men. you can group this logic